src/VisualStudio/Core/Def/Implementation/PreviewPane/PreviewPaneService.cs

```diff
@@ -65,29 +65,29 @@ private Image GetSeverityIconForDiagnostic(Diagnostic diagnostic)
 
         object IPreviewPaneService.GetPreviewPane(Diagnostic diagnostic, object previewContent)
         {
-            var telemetry = diagnostic == null ? false : diagnostic.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.Telemetry);
+            var isHeaderMissing = (diagnostic == null) || diagnostic.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.Trigger);
 
-            if ((diagnostic == null) && (previewContent == null))
+            if (isHeaderMissing)
             {
-                // Bail out in cases where there is no diagnostic (which means there is nothing to put in
-                // the header section of the preview pane) as well as no preview content (i.e. no diff view).
-                return null;
-            }
+                if (previewContent == null)
+                {
+                    // Bail out in cases where there is nothing to put in the header section
+                    // of the preview pane and no preview content (i.e. no diff view) either.
+                    return null;
+                }
 
-            if ((diagnostic == null) || (diagnostic.Descriptor is TriggerDiagnosticDescriptor))
-            {
                 return new PreviewPane(
-                    null, null, null, null, null, null, telemetry, previewContent, _serviceProvider);
-            }
-            else
-            {
-                return new PreviewPane(
-                    GetSeverityIconForDiagnostic(diagnostic),
-                    diagnostic.Id, diagnostic.GetMessage(),
-                    diagnostic.Descriptor.MessageFormat.ToString(DiagnosticData.USCultureInfo),
-                    diagnostic.Descriptor.Description.ToString(CultureInfo.CurrentUICulture),
-                    diagnostic.Descriptor.HelpLinkUri, telemetry, previewContent, _serviceProvider);
+                    null, null, null, null, null, null, false, previewContent, _serviceProvider);
```

Comment:
I suppose you were asking to use named args for the nulls @srivatsn. I'll do that. 

Another constructor overload would be nice. The number of arguments has grown over time as we added various functionality. I started refactoring the code in PreviewPane.xaml.cs to introduce another constructor and extract out helpers to initialize various portions of the UI - but this ended up taking more time than I would like to spend on this now. I'll refactor this the next time I am in PreviewPane.xaml.cs :)